pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/geospatial/H3IndexResolution.java

```diff
@@ -0,0 +1,53 @@
+package org.apache.pinot.core.segment.creator.impl.geospatial;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Stores the resolutions for an index. There are in total of H3 resolutions https://h3geo.org/#/documentation/core-library/resolution-table
+ * To efficiently serialize the resolutions, we use two bytes for encoding th enabled resolutions. The resolution level
+ * maps to the corresponding bit.
+ */
+public class H3IndexResolution {
+  private short _resolutions;
```

Comment:
use INT

Reply:
I feel short is easier since all resolutions can be represented in two bytes, so I don't have to reason about the first leftmost two bytes.